src/libs/Localize/index.ts

```diff
@@ -152,7 +152,7 @@ function translate<TKey extends TranslationPaths>(desiredLanguage: 'en' | 'es' |
     const languageAbbreviation = desiredLanguage.substring(0, 2) as 'en' | 'es';
 
     const translatedPhrase = getTranslatedPhrase(language, phraseKey, languageAbbreviation, ...phraseParameters);
-    if (translatedPhrase) {
+    if (translatedPhrase !== null && translatedPhrase !== undefined) {
```

Comment:
@fedirjh Your latest code looks nice and I have incorporated it with just renaming the type name from `CacheValue` to `TranslatedValue`. Is that fine with you?

> I believe the correct way to fix this bug is to return the result of the translation as it is, similar to what was previously done

If we did that we will have the same bug where we show the translation key in UI for the value which is an empty string. To fill you in, previously we were checking for Truthiness of `translatedPhrase` and if it's an empty string, which can happen because of variables, we will have an empty string as `translatedPhrase`, which is what is expected. So in such cases, returning `phraseKey` instead would be a violation.

But I guess it's also fixed by your latest code, so that's fine 👍

Reply:
I actually think that the code was simpler and just as clear before - strings (including empty strings) are valid results, but `null` or `undefined` are not.

Reply:
i.e: if the translation isn't found, `getTranslatedPhrase` should just return `null`, rather than `{value: '', isValidKey: false}`. If the translation is found and happens to be an empty string, that's fine